dll/win32/browseui/aclistisf.cpp

```diff
@@ -20,83 +21,411 @@
+    return S_OK;
+}
+
 // *** IEnumString methods ***
-HRESULT STDMETHODCALLTYPE CACListISF::Next(ULONG celt, LPOLESTR *rgelt, ULONG *pceltFetched)
+STDMETHODIMP CACListISF::Next(ULONG celt, LPOLESTR *rgelt, ULONG *pceltFetched)
 {
-    TRACE("(%p, %d, %p, %p)\n", this, celt, rgelt, pceltFetched);
-    return E_NOTIMPL;
+    TRACE("Next(%p, %d, %p, %p)\n", this, celt, rgelt, pceltFetched);
+
+    if (celt == 0)
+        return S_OK;
+    if (!rgelt)
+        return S_FALSE;
+
+    *rgelt = NULL;
+    if (pceltFetched)
+        *pceltFetched = 0;
+
+    if (!m_pEnumIDList)
+    {
+        NextLocation();
+        if (!m_pEnumIDList)
+            return S_FALSE;
+    }
+
+    HRESULT hr;
+    CComHeapPtr<ITEMIDLIST> pidlChild;
+    CComHeapPtr<WCHAR> pszPathName;
+
+    do
+    {
+        for (;;)
+        {
+            pidlChild.Free();
+            hr = m_pEnumIDList->Next(1, &pidlChild, NULL);
+            if (hr != S_OK)
+                break;
+
+            pszPathName.Free();
+            GetPathName(pidlChild, pszPathName);
+            if (!pszPathName)
+                continue;
+
+            if (m_dwOptions & (ACLO_FILESYSONLY | ACLO_FILESYSDIRS))
+            {
+                DWORD attrs = SFGAO_FILESYSANCESTOR | SFGAO_FILESYSTEM;
+                hr = m_pShellFolder->GetAttributesOf(1, const_cast<LPCITEMIDLIST *>(&pidlChild), &attrs);
+                if (SUCCEEDED(hr))
+                {
+                   if (!(attrs & (SFGAO_FILESYSTEM | SFGAO_FILESYSANCESTOR)))
+                        continue;
+                }
+            }
+
+            if ((m_dwOptions & ACLO_FILESYSDIRS) && !PathIsDirectoryW(pszPathName))
+                continue;
+
+            hr = S_OK;
+            break;
+        }
+    } while (hr == S_FALSE && NextLocation() == S_OK);
+
+    if (hr == S_OK)
+    {
+        *rgelt = pszPathName.Detach();
+        if (pceltFetched)
+            *pceltFetched = 1;
+    }
+    else
+    {
+        hr = S_FALSE;
+    }
+
+    TRACE("*rgelt: %S\n", *rgelt);
+    return hr;
 }
 
-HRESULT STDMETHODCALLTYPE CACListISF::Reset()
+STDMETHODIMP CACListISF::Reset()
 {
-    TRACE("(%p)\n", this);
-    return E_NOTIMPL;
+    TRACE("Reset(%p)\n", this);
+
+    m_iNextLocation = LT_DIRECTORY;
+    m_szExpand = L"";
+
+    SHELLSTATE ss = { 0 };
+    SHGetSetSettings(&ss, SSF_SHOWALLOBJECTS, FALSE);
+    m_fShowHidden = ss.fShowAllObjects;
+
+    if (m_dwOptions & ACLO_CURRENTDIR)
+    {
+        CComHeapPtr<ITEMIDLIST> pidl;
+        if (m_pBrowserService)
+        {
+            m_pBrowserService->GetPidl(&pidl);
+            if (pidl)
+                Initialize(pidl);
+        }
+        HRESULT hr = SetLocation(pidl);
+        if (FAILED(hr))
+            return S_FALSE;
+    }
+    return S_OK;
 }
 
-HRESULT STDMETHODCALLTYPE CACListISF::Skip(ULONG celt)
+STDMETHODIMP CACListISF::Skip(ULONG celt)
 {
-    TRACE("(%p, %d)\n", this, celt);
+    TRACE("Skip(%p, %d)\n", this, celt);
```

Comment:
If you set the environment variable `DEBUGFORMAT=wine` the function name will be included.

Reply:
The function names in TRACE are removed. OK.